ledger-core/virtual/execute/execute.go

```diff
@@ -810,15 +757,8 @@ func (s *SMExecute) stepSaveNewObject(ctx smachine.ExecutionContext) smachine.St
 		}
 	}
 
-	switch s.objectSharedState.Prepare(action).TryUse(ctx).GetDecision() {
-	case smachine.NotPassed:
-		return ctx.WaitShared(s.objectSharedState.SharedDataLink).ThenRepeat()
-	case smachine.Impossible:
-		panic(throw.NotImplemented())
-	case smachine.Passed:
-		// go further
-	default:
-		panic(throw.Impossible())
+	if stepUpdate, ok := s.shareObjectAccess(ctx, action); !ok {
```

Comment:
You can use simple `stepUpdate.IsEmpty()` instead of second argument.

Reply:
Sound good